### DISCUSSION TITLE
Bug in OB1 with G6

### DISCUSSION THREAD

**Discussion by Miquest** [NightscoutFoundation/xDrip#2182] [Q&A]
Hello,
I upgraded to OxygenOS 12 on my OnePlus 8T and since then, the OB1 Collector goes always in the background, except if I use the phone. I deactivated the Doze Mode completely via ADB, and my Battery Settings are also correct. It does work with the recommended Settings, but only every 10 minutes. Is it possible to wake up xDrip every 4 minutes to get readings?

**Comment by Navid200** [comment]
@psonnera

**Reply by psonnera** [reply]
I've been thinking about it a lot but didn't come with an intelligent answer. The only idea that went through my mind was to setup a silent alarm triggering every 1 or 2 minutes to check the issue is due to xDrip+ falling asleep or something else. The annoying side effect is a permanent flow of notifications, but that's only a test.

**Reply by Miquest** [reply]
How can I do this? Or is it a new function which could be implemented?

**Reply by Miquest** [reply]
If I restart the OB1 Collector manually, it's working perfectly...

**Reply by Miquest** [reply]
Yes, the G6

**Reply by Navid200** [reply]
It really helps if you include that in your post instead of OB1.
I don't know why I thought this was about Libre and that's why I tagged Patrick.

Are all your xDrip and Android settings in agreement with this?
https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html

**Reply by Miquest** [reply]
Sorry, I thought OB1 is only for G6...
I will change the heading.
Yes, all my settings agree with this page...
I think it's a bug...

**Reply by Navid200** [reply]
It is.  
But, I look for G6 or G5 or Dexcom.  I hardly ever look for OB1.
Not everyone knows everything there is about all xDrip capabilities.  I don't know much about Libre.  If I don't recognize any expected phrases, I may assume it is about Libre and skip the thread or tag someone else who may know about Libre.

**Reply by Navid200** [reply]
A bug is something that affects almost everyone.
Intermittent connectivity is a problem that many experience and we are improving the code to fix the obvious problems.

Please post screenshots of the following:
Settings &#8722;> G5/G6 Debug Settings  
Settings &#8722;> Less common Settings &#8722;> Bluetooth Settings  
Settings &#8722;> Less common settings &#8722;> Other misc options

**Reply by Miquest** [reply]
![Screenshot_2022-06-25-13-28-47-45](https://user-images.githubusercontent.com/91971937/175771600-1458af16-30cb-4140-a5a6-dda77f42edc4.jpg)

**Reply by Miquest** [reply]
![Screenshot_2022-06-25-13-29-08-11](https://user-images.githubusercontent.com/91971937/175771615-fcab162d-720a-4864-a236-6b4be879498a.jpg)

**Reply by Miquest** [reply]
![Screenshot_2022-06-25-13-29-24-70](https://user-images.githubusercontent.com/91971937/175771623-caf7bf91-f7f6-4d3f-bea3-7e9550502f55.jpg)

**Reply by Navid200** [reply]
Looks like you have disabled Bluetooth watchdog.
That's not in agreement with the recommended settings.

Please also post a screenshot of the system status page.  You can blackout the transmitter ID.

**Reply by Miquest** [reply]
![Screenshot_2022-06-25-13-42-06-83_2dd2c1bcb8f659cd696bcba2072f751b](https://user-images.githubusercontent.com/91971937/175772057-33926c31-bc75-4dd7-8c5a-ef2a150dc387.jpg)


I have to disable Avoid Scanning and minimize Scanning...
Otherwise I cannot receive readings

**Reply by psonnera** [reply]
Disable trust auto-connect, see if it improves then disable background scans in xDrip+ bluetooth settings.

**Reply by Miquest** [reply]
I disabled it, but nothing happened...

**Comment by Navid200** [comment]
@thorinEichenschild007 OB1?
Do you use a Dexcom device?
